aten/src/ATen/native/Indexing.cpp

```diff
@@ -272,7 +272,10 @@ Tensor & index_copy_(Tensor & self, int64_t dim, const Tensor & index, const Ten
   int64_t numIndices = index.numel();
   if (source.dim() == 0 && numIndices != 1) {
     AT_INDEX_ERROR("index_copy_(): When source is scalar, index should have one element (got ", numIndices, ")");
+  } else if (source.dim() != self.dim()) {
+    AT_INDEX_ERROR("index_copy_(): When source is not scalar, its dimention must match destinations");
```

Comment:
/dimention/dimension/

Reply:
Maybe: `When source is not scalar, its dimensionality must match the dimensionality of the destination`